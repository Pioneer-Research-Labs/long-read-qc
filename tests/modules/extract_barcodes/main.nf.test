nextflow_process {

    name "Test Process extract_barcodes"
    script "modules/extract_barcodes/main.nf"
    process "extract_barcodes"

    test("Should extract barcodes") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = Channel.fromList([ tuple( [id:'Gateway_EmptyBarcodedLib1_20241217',
                genome:'H_elongata'],
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/5HKJT6_1_sample_1.fastq.gz'),
                file('${projectDir}test_data/Gateway_EmptyBarcodedLib1_20241217/c.00323.dna'),
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/flanking.gb'))])
                """
            }
        }

        then {
            assert process.success
            assert path("$launchDir/tests/results/Gateway_EmptyBarcodedLib1_20241217/barcodes.fasta").readLines().size() == 12560
            assert snapshot("$launchDir/results/Gateway_EmptyBarcodedLib1_20241217/barcodes.fasta").match()
        }

    }
        test("Should extract 0 barcodes") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = Channel.fromList([ tuple( [id:'Gib_Lib1_240801',
                genome:'H_elongata'],
                file('${projectDir}/test_data/Gib_Lib1_240801/M49T52_1_m.C.00160.fastq'),
                file('${projectDir}test_data/Gib_Lib1_240801/c.00203.dna'),
                file('${projectDir}/test_data/Gib_Lib1_240801/flanking.gb'))])
                """
            }
        }

        then {
            assert process.success
            assert process.out.size() == 2
            assert process.out[0][0].size() == 2
            assert process.out[1].size() == 1
            assert path("$launchDir/tests/results/Gib_Lib1_240801/barcodes.fasta").readLines().size() == 0
            assert snapshot("$launchDir/results/Gib_Lib1_240801/barcodes.fasta").match()
        }

    }
    test("Should extract barcodes from a sample with no inserts") {
        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = Channel.fromList([ tuple( [id:'b.33TransposonBarcodedLibraryLinear',
                genome:'H_elongata'],
                file('${projectDir}/test_data/b.33TransposonBarcodedLibraryLinear/43QQ8C_3_PCR_440.fastq'),
                file('${projectDir}test_data/b.33TransposonBarcodedLibraryLinear/c.00410.dna'),
                file('${projectDir}/test_data/b.33TransposonBarcodedLibraryLinear/flanking.gb'))])
                """
            }
        }

        then {
            assert process.success
            assert path("$launchDir/tests/results/b.33TransposonBarcodedLibraryLinear/barcodes.fasta").readLines().size() == 1464
            assert snapshot("$launchDir/results/b.33TransposonBarcodedLibraryLinear/barcodes.fasta").match()
        }
   }

}
