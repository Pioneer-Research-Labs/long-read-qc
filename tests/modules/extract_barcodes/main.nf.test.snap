{
    "Should extract barcodes": {
        "content": [
            "/Users/sandie/projects/pioneerlabs/long-read-qc/.nf-test/tests/5bd742095678ab608393c0833c02e35b/results/Gateway_EmptyBarcodedLib1_20241217/barcodes.fasta"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-03-31T15:29:07.029881"
    },
    "Should extract 0 barcodes": {
        "content": [
            "/Users/sandie/projects/pioneerlabs/long-read-qc/.nf-test/tests/fcf2849eeeb2df1daef7c9b523092dce/results/Gib_Lib1_240801/barcodes.fasta"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-03T13:59:56.415429"
    },
    "Should extract barcodes from a sample with no inserts": {
        "content": [
            "/Users/sandie/projects/pioneerlabs/long-read-qc/.nf-test/tests/2ef1d4bc6cd917a0e05f55beb38081c0/results/b.33TransposonBarcodedLibraryLinear/barcodes.fasta"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-03T14:19:32.683705"
    }
}